source/utilities.cc

```diff
@@ -95,225 +195,111 @@ namespace aspect
-                }
-            }
+            if (check_structure)
+              {
+                AssertThrow((*n_values_per_key)[field_index] == n_values,
+                            ExcMessage("The key <" + field_names[field_index] + "> in <"+ property_name + "> does not have "
+                                       + "the expected number of values. It expects " + std::to_string((*n_values_per_key)[field_index])
+                                       + " values, but we found " + std::to_string(n_values) + " values."));
 
-          for (const auto &entry: return_map)
-            for (const auto &sub_entry: entry)
-              return_values.push_back(sub_entry);
-        }
-      else if (Patterns::List(Patterns::Double(),1,n_fields).match(input_string))
-        {
-          // Handle the format of a comma separated list of doubles, with no keywords
-          return_values = possibly_extend_from_1_to_N (dealii::Utilities::string_to_double(dealii::Utilities::split_string_list(input_string)),
-                                                       n_fields,
-                                                       property_name);
+              }
 
-          if (store_structure)
-            {
-              for (unsigned int i=0; i<n_fields; ++i)
-                (*n_values_per_key)[i] = 1;
-            }
+            ++field_index;
+          }
+      }
 
-          if (check_structure)
-            {
-              for (unsigned int i=0; i<n_fields; ++i)
-                AssertThrow((*n_values_per_key)[i] == 1,
-                            ExcMessage("The input parameter " + property_name + " seems to consist of a list "
-                                       "of doubles. In this case we only allow a single value per provided key, but the "
-                                       "assumed structure that was provided to the function expects " + std::to_string((*n_values_per_key)[i]) +
-                                       " values for key " + field_names[i] + ". To specify more than one value per field for this input parameter, "
-                                       "you need to use the format " + field_names[i] + ":value1|value2|..."));
-            }
-        }
-      else
+      // Finally: Convert the map into a vector of doubles, sorted in the order
+      // of the field_names input parameter
+      std::vector<double> return_values;
+      for (const auto &field_name: field_names)
         {
-          // No Patterns matches were found!
-          AssertThrow (false,
-                       ExcMessage ("The required format for field <"
-                                   + property_name
-                                   + "> was not found. Specify a comma separated "
-                                   + "list of `<double>' or `<key1> : <double>|<double>|..., "
-                                   + "<key2> : <double>|... , ... '."));
+          const auto entry_range = parsed_map.equal_range(field_name);
```

Comment:
Can you type out the variable type of `entry_range`? I find this really hard to read... I hadn't seen the equal_range function before, but if you tell me that the return type is a pair of iterators, it would be more clear what's happening.